File: C/Lei_de_Ohm.c
#include <stdio.h>

float PdadoVI(float pV, float pI){
	return pV*pI;
}

float IdadoVR(float pV, float pR){
	return pV/pR;
}

float RdadoVP(float pV, float pI){
 	return pV/pI;
}

float RdadoVR(float pV, float pP){
 	return (pV*pV)/pP;
}

int main(){
  	int opcao = 0;
  	float V,R,P,I;
  
 	while(1){
  
    		printf("1- Digite 1 para calcular P dados V e I\n");
    		printf("2- Digite 2 para calcular I dados V e R\n");
    		printf("3- Digite 3 para calcular R dados P e V\n");
    		printf("4- Digite 4 para calcular R dados V e I\n");
    		printf("5- Digite 5 para SAIR\n\n");
    		scanf("%d",&opcao);
 
  		switch (opcao){
  			case 1:
    				printf("Digite V:\n");
    				scanf("%f",&V);
    				printf("Digite I:\n");
				scanf("%f",&I);
				P = PdadoVI (V,I);
				printf("Valor de P = %f\n",P);
				break;
			case 2:
				printf("Digite V:\n");
				scanf("%f",&V);
				printf("Digite R:\n");
				scanf("%f",&R);
				I = IdadoVR (V,R);
				printf("Valor de I = %f\n",I);
				break;
			case 3:
				printf("Digite V:\n");
				scanf("%f",&V);
				printf("Digite P:\n");
				scanf("%f",&P);
				R = RdadoVP (V,P);
				printf("Valor de R = %f\n",R);
				break;
			case 4:
				printf("Digite V:\n");
				scanf("%f",&V);
				printf("Digite I:\n");
				scanf("%f",&I);
				R = RdadoVR (V,I);
				printf("Valor de R = %f\n",R);
				break;
			case 5:
				printf("Encerrando a aplicação\n");
				return -1;
			default:
				printf("Opção inválida\n\n");
			}
	}
	return 0;
}

File: Python/Lei_de_Ohm.py
class Lei_de_Ohm:


	def __init__(self):
		print("\n1- Digite 1 para calcular P dados V e I")
		print("2- Digite 2 para calcular I dados V e R")
		print("3- Digite 3 para calcular R dados P e V")
		print("4- Digite 4 para calcular R dados V e I")
		opcao = int(input("5- Digite 5 para SAIR\n\n"))
		
		if opcao == 1:
			V = float(input( "\nDigite V: "))
			I = float(input( "Digite I: "))
			P = self.PdadoVI(V, I)
			print("\nO valor de P é: ", P)
			
		elif opcao == 2:
			V = float(input( "\nDigite V: "))
			R = float(input( "Digite R: "))
			I = self.IdadoVR(V, R)
			print("\nO valor de I é: ", I)
			
		elif opcao == 3:
			V = float(input( "\nDigite V: "))
			P = float(input( "Digite P: "))
			R = self.RdadoVP(V, P)
			print("\nO valor de R é: ", R)
			
		elif opcao == 4:
			V = float(input( "\nDigite V: "))
			I = float(input( "Digite I: "))
			I = self.RdadoVI(V, I)
			print("\nO valor de R é: ", R)
			
		elif opcao == 5:
			pass
			
		else:
			print("\nOpção inválida! \n\n")
			repete = Lei_de_Ohm()
			
	def PdadoVI(self, v, i):
		return v*i
		
	def IdadoVR(self, v, r):
		return v/r
	
	def RdadoVP(self, v, p):
		return (v*v)/p
		
	def RdadoVI(self, v, i):
		return v/i
		
inicio = Lei_de_Ohm()
